src/hooks/useBootstrapApp.ts

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from 'react-query';
+
+import { CACHE_TIME, STALE_TIME } from '#src/config';
+import { getModule } from '#src/container';
+import AppController from '#src/stores/AppController';
+import { useTrackConfigKeyChange } from '#src/hooks/useTrackConfigKeyChange';
+import { SettingsController } from '#src/stores/SettingsController';
+import type { Config } from '#types/Config';
+import type { Settings } from '#src/stores/SettingsStore';
+
+const applicationController = getModule(AppController);
+const settingsController = getModule(SettingsController);
+
+type Resources = {
+  config: Config;
+  configSource: string | undefined;
+  settings: Settings;
+};
+
+export const useBootstrapApp = (onReady: () => void) => {
+  const { data, isLoading, error, isError, isSuccess } = useQuery<Resources, Error>(
+    'config-init',
+    async () => {
+      const settings = await settingsController.initialize();
```

Comment:
This is a nice simplification of the multiple queries into 1 sequential flow. I think you'll still have the problem of identifying a settings error from a config error, but that can be handled by seeing if settings is undefined or config is undefined.

Reply:
Yes, that's correct. We can also use custom errors to make it more concrete.

Reply:
My preference would be the former instead of creating custom error classes. For just this use case that seems like overkill.

Reply:
Agree!

Reply:
Actually, there already are custom errors: `ConfigError` and `SettingsError` 😅

Reply:
Yeah, that is what I started with. We have discussed it here:

https://github.com/jwplayer/ott-web-app/pull/363#discussion_r1386553458

We can use `undefined` for config / settings, but then we won't able to differentiate between different config's errors (is it missing or is api broken?). Though it is maybe not as important.

Reply:
Ah, yes, I see... There is even more discussion overlap. Sorry about that!